### PR TITLE
[Core/Plugin] Athena Framework Fingerpointing

### PR DESCRIPTION
**Pull request by iCry1337**
Just would have your feedback and implement this as a possible inbuild plugin.

Credits to : https://github.com/tastydev/fingerpointing-altv

### CODE REVIEW

File: src/core/plugins/core-fingerpoint/server/index.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import * as alt from 'alt-server';
+import { PluginSystem } from '../../../server/systems/plugins';
```
Comment (Stuyk) [inline src/core/plugins/core-fingerpoint/server/index.ts:2]: You can actually remove this entire file. It's only necessary to register a plugin on server-side.

**Review by jonesXYZ — COMMENTED** [review]
Please follow the contributing guidelines

File: src/core/plugins/core-fingerpoint/client/index.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
+import * as alt from 'alt-client';
+import * as natives from 'natives';
+
+let FINGERPOINT_KEY = 66
```
Comment (Stuyk) [inline src/core/plugins/core-fingerpoint/client/index.ts:4]: Probably rename this to something more meaningful. Such as `KEYBOARD_B_KEY`.

File: src/core/plugins/core-fingerpoint/client/index.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+export class Fingerpointing {
```
Comment (Stuyk) [inline src/core/plugins/core-fingerpoint/client/index.ts:6]: Doesn't actually need to be a class at all. Can just be a const object.

See: #299

File: src/core/plugins/core-fingerpoint/client/index.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            if (key !== FINGERPOINT_KEY) return;
+            this.stop();
+        });
+    }
+
+    async start() {
+        if (this.active) return;
+        this.active = true;
+        try {
+            await this.requestAnimDictPromise('anim@mp_point')
```
Comment (Stuyk) [inline src/core/plugins/core-fingerpoint/client/index.ts:31]: Can use internal animation loader.

File: src/core/plugins/core-fingerpoint/client/index.ts [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+    isBlockingAllowed() {
+        const isAllowed = Date.now() - this.lastBlockDate > this.debounceTime;
+        if (isAllowed) {
+            this.lastBlockDate = null;
+        }
+        return isAllowed;
+    }
+
+    requestAnimDictPromise(dict) {
```
Comment (Stuyk) [inline src/core/plugins/core-fingerpoint/client/index.ts:179]: Can't be removed entirely. There's animation loading built into Athena.

File: src/core/plugins/core-fingerpoint/client/index.ts [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+export class Fingerpointing {
+
+    active: boolean = false;
+    interval: number = null;
+    cleanStart: boolean = false;
+    debounceTime: number = 150;
+    lastBlockDate: number = null;
+    localPlayer = alt.Player.local;
+
+    registerEvents() {
```
Comment (Stuyk) [inline src/core/plugins/core-fingerpoint/client/index.ts:15]: Nothing ever calls this to register the keybinds.

File: src/core/plugins/core-fingerpoint/client/index.ts [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        alt.on('keydown', (key) => {
```
Comment (Stuyk) [inline src/core/plugins/core-fingerpoint/client/index.ts:16]: Could use Athena's internal keybind system such as `keyHeld`

**Review by Stuyk — CHANGES_REQUESTED** [review]

File: src/core/plugins/core-fingerpoint/client/index.ts (hunk 4 above)
Comment (iCry1337) [reply src/core/plugins/core-fingerpoint/client/index.ts:31]: Some references/examples to it?

**Pull request closed without merging**